tarantool/crud.py

```diff
@@ -0,0 +1,87 @@
+"""
+This module provides API for interaction with the `crud`_ module.
+
+.. _crud: https://github.com/tarantool/crud/
+"""
+
+from tarantool.error import DatabaseError
+
+
+class CrudResponse(object):
+    """
+    Contains response fields from the `crud`_ module that correspond 
+    to the lua implementation.
+
+    .. _crud: https://github.com/tarantool/crud/
+    """
+
+    def __init__(self, response):
+        """
+        Sets response fields as in lua implementation.
+
+        :param response: The response object of the crud module call.
+        :type response: :class:`~tarantool.response.Response`
+        """
+
+        if isinstance(response, dict):
+            for response_field_name in response.keys():
+                if isinstance(response_field_name, bytes):
+                    setattr(self, response_field_name.decode(), response[response_field_name])
+                else:
+                    setattr(self, response_field_name, response[response_field_name])
+        elif isinstance(response, str) or isinstance(response, bytes):
+            self.str = response
+            self.err = response
+        else:
+            raise RuntimeError('Unable to decode response to object due to unknown type')
+
+
+class CrudResult(CrudResponse):
+    """
+    Contains result's fields from result variable 
+    of crud module operation.
+    """
+
+
+class CrudError(CrudResponse):
+    """
+    Contains error's fields from error variable 
+    of crud module operation.
+    """
+
+
+def wrapCrudUndefinedProcedureError(e: DatabaseError):
+    """
+    Supplements information about an unsuccessful attempt 
+    to find the crud module procedure.
+
+    :param e: The error object of the crud module call.
+    :type e: :class:`~tarantool.error.DatabaseError`
+    """
+
+    info = ". Ensure that you're calling crud.router and user have enough grants"
+    e.message = e.message + info
+    e.extra_info.message = e.extra_info.message + info
+    e.args = (e.args[0], e.args[1] + info)
+
+    return e
+
+def call_crud(conn, *args):
+    """
+    Calls the crud via connection.call with try/except block.
+
+    :param conn: The connection object for the crud module call.
+    :type conn: :class:`~tarantool.connection.Connection`
+
+    :param args: The arguments to pass to the crud module.
+    :type args: :obj:`tuple`
+    """
+
+    try:
+        crud_resp = conn.call(*args)
+    except DatabaseError as e:
+        if e.code == 33:
```

Comment:
Let's use a named constant here instead of `33` (unfortunately, current `error.py` is a bit messy)